### PR TITLE
Install.js: Add a check for mongodb replset initiation that doesn't actually do anything besides alert the user

### PR DESCRIPTION
**Pull request by zamabe**
**Please do not merge this, I've done zero testing**

Added a step before in the mongodb install stage, before it is declared finished, that waits for it to finish doing replSet initiation.

- I've used `process` and `time` without knowing if they need to be `require()`'d.
- I've used `time.sleep` without knowing if it's available. I assume it blocks and accepts integer and float numbers as seconds.
- I have no idea how `rs.status` should actually be called in mongo nor where to find `myState` in `result`, so I guessed.

### CODE REVIEW

**Comment by rickihastings** [comment]
Thanks for this! I'll give it a test tomorrow and feed back.

File: install.js [hunk 1]
```diff
@@ -197,6 +197,30 @@ function isMongoRunning(isGlobal, path) {
 		});
 	}
 
+	// waittime is in seconds
+	function waitForRsInitiated(db, originalResult, waittime=30) {
+		waittime = waittime * 2 // check every half-second, so double the count
+
+		process.stdout.write(COLOUR.blue, 'Checking if the replica set is fully initiated...');
+		while (waittime > 0) {
```
Comment (zamabe) [inline install.js]:
```diff
-while (waittime > 0) {
+while (waittime => 0) {
```

If it doesn't match 0 it will never print the warning

File: install.js [hunk 2]
```diff
@@ -197,6 +197,30 @@ function isMongoRunning(isGlobal, path) {
...
+					process.stdout.write(' Success!\n');
+					break;
+				} else if (waittime > 0) {
+					waittime = waittime - 1;
+					process.stdout.write('.');
+					time.sleep(0.5);
+				} else {
+					process.stdout.write('\n');
+					console.log(COLOUR.yellow, 'It appears that the mongodb replica set has not yet been initiated.');
+					console.log(COLOUR.yellow, 'MongoDB may magically correct this in seconds or minutes, but please be aware of this issue.');
```
Comment (zamabe) [inline install.js]:
```diff
 					console.log(COLOUR.yellow, 'MongoDB may magically correct this in seconds or minutes, but please be aware of this issue.');
+					break;
```

Break the loop once we're here so it doesn't loop again and 0-1 a potentially unsigned int before it terminates anyway.

File: install.js [hunk 3]
```diff
@@ -197,6 +197,30 @@ function isMongoRunning(isGlobal, path) {
...
+				}
```
Comment (zamabe) [inline install.js]:
```diff
 				if (result.myState == 1) {
 					process.stdout.write(' Success!\n');
 					break;
-				} else if (waittime > 0) {
-					waittime = waittime - 1;
-					process.stdout.write('.');
-					time.sleep(0.5);
-				} else {
+				} else if (waittime == 0) {
 					process.stdout.write('\n');
 					console.log(COLOUR.yellow, 'It appears that the mongodb replica set has not yet been initiated.');
 					console.log(COLOUR.yellow, 'MongoDB may magically correct this in seconds or minutes, but please be aware of this issue.');
+					break;
+				} else {
+					waittime = waittime - 1;
+					process.stdout.write('.');
+					time.sleep(0.5);
 				}
```

Break the loop once we're here so it doesn't loop again and 0-1 a potentially unsigned int before the loop just terminates anyway on the next iteration.

**Pull request merged**